modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectRelationship/ObjectRelationshipFormBase.tsx

```diff
@@ -136,13 +136,15 @@ export function ObjectRelationshipFormBase({
 	const [query, setQuery] = useState<string>('');
 
 	const [types, selectedType] = useMemo(() => {
-		const types = [ONE_TO_MANY, MANY_TO_MANY];
-		if (ffOneToOneRelationshipConfigurationEnabled) {
-			types.push(ONE_TO_ONE);
-		}
-
-		return [types, types.find(({value}) => value === values.type)?.label];
-	}, [ffOneToOneRelationshipConfigurationEnabled, values.type]);
+		const types = [MANY_TO_MANY, ONE_TO_MANY, ONE_TO_ONE];
+
+		return [
+			types.filter((relationshipType) =>
+				objectRelationshipTypes?.includes(relationshipType.value)
+			),
+			types.find(({value}) => value === values.type)?.label,
+		];
+	}, [objectRelationshipTypes, values.type]);
```

Comment:
Hey @murilostodolni we have 2 performance issues here, first every time this `useMemo` is executed we are creating this array of `types` in line `139` again, no need as it is a static array, we can create it outside the `ObjectRelationshipFormBase` component so it will be created only once instead, you can create it just below the `ONE_TO_ONE` const declaration and name it `OBJECT_RELATIONSHIP_TYPES`


```tsx
const OBJECT_RELATIONSHIP_TYPES = [MANY_TO_MANY, ONE_TO_MANY, ONE_TO_ONE];
```

The second problem is that every time the `values.types` change we are re-filtering the `types` array on lines `142-144` unnecessarily, we can refactor this `useMemo` to just do the `types` filter and put the `selectedType` filter directly on the `SingleSelect value` :

```tsx
	const types = useMemo(() => {
		return OBJECT_RELATIONSHIP_TYPES.filter((relationshipType) =>
			objectRelationshipTypes?.includes(relationshipType.value)
		);
	}, [objectRelationshipTypes]);
```
```tsx
	<SingleSelect
		disabled={readonly}
		error={errors.type}
		label={Liferay.Language.get('type')}
		onChange={({value}) => setValues({type: value})}
		options={types}
		required
		value={types.find(({value}) => value === values.type)?.label}
	/>
```